streaming/base/converters/dataframe_to_mds.py

```diff
@@ -82,25 +83,36 @@ def map_spark_dtype(spark_data_type: Any) -> str:
         Raises:
             raise ValueError if no mds datatype is found for input type
         """
-        mds_type = MAPPING_SPARK_TO_MDS.get(type(spark_data_type), None)
+        if issubclass(type(spark_data_type), DecimalType):
+            mds_type = SPARK_TO_MDS.get(DecimalType(), None)
+        else:
+            mds_type = SPARK_TO_MDS.get(spark_data_type, None)
+
         if mds_type is None:
-            raise ValueError(f'{spark_data_type} is not supported by MDSWriter')
+            raise ValueError(f'{spark_data_type} is not supported by dataframe_to_mds')
         return mds_type
 
     # user has provided schema, we just check if mds supports the dtype
     if user_defined_cols is not None:
         mds_supported_dtypes = {
-            mds_type for mds_type in MAPPING_SPARK_TO_MDS.values() if mds_type is not None
+            mds_type for mds_type in SPARK_TO_MDS.values() if mds_type is not None
         }
         for col_name, user_dtype in user_defined_cols.items():
             if col_name not in dataframe.columns:
                 raise ValueError(
                     f'{col_name} is not a column of input dataframe: {dataframe.columns}')
+            if user_dtype.startswith('ndarray'):
+                user_dtype = ':'.join(user_dtype.split(':')[:-1])
```

Comment:
i'm confused and suspicious about what's going here. are we just dropping the shape constraint if present? needs to handle the three cases like this right?

`ndarray` -> `ndarray`
`ndarray:dtype` -> `ndarray:dtype`
`ndarray:dtype:shape` -> `ndarray:dtype`

Reply:
Good point. It is actually does not include ndarray -> ndarray , I made change so it includes the latter two.